section-05-production-model-package/regression_model/pipeline.py

```diff
@@ -0,0 +1,125 @@
+from feature_engine.encoding import (
+    RareLabelEncoder,
+    OrdinalEncoder,
+)
+
+# from feature-engine
+from feature_engine.imputation import (
+    AddMissingIndicator,
+    MeanMedianImputer,
+    CategoricalImputer,
+)
+from feature_engine.selection import DropFeatures
+from feature_engine.transformation import LogTransformer
+from feature_engine.wrappers import SklearnTransformerWrapper
+from sklearn.linear_model import Lasso
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import MinMaxScaler, Binarizer
+
+from regression_model.config.core import config
+from regression_model.processing import preprocessors as pp
+
+
+price_pipe = Pipeline(
+    [
+        # ===== IMPUTATION =====
+        # impute categorical variables with string missing
+        (
+            "missing_imputation",
+            CategoricalImputer(
+                imputation_method="missing",
+                variables=config.model_config.categorical_vars_with_na_missing,
+            ),
+        ),
+        (
+            "frequent_imputation",
+            CategoricalImputer(
+                imputation_method="frequent",
+                variables=config.model_config.categorical_vars_with_na_frequent,
+            ),
+        ),
+        # add missing indicator
+        (
+            "missing_indicator",
+            AddMissingIndicator(variables=config.model_config.numerical_vars_with_na),
+        ),
+        # impute numerical variables with the mean
+        (
+            "mean_imputation",
+            MeanMedianImputer(
+                imputation_method="mean",
+                variables=config.model_config.numerical_vars_with_na,
+            ),
+        ),
+        # == TEMPORAL VARIABLES ====
+        (
+            "elapsed_time",
+            pp.TemporalVariableTransformer(
+                variables=config.model_config.temporal_vars,
+                reference_variable=config.model_config.ref_var,
+            ),
+        ),
+        ("drop_features", DropFeatures(features_to_drop=[config.model_config.ref_var])),
```

Comment:
here, we are using `ref_var` instead of `drop_features`, so we don't need drop_features in the config.